maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/FileExtractor.java

```diff
@@ -0,0 +1,133 @@
+package org.jboss.shrinkwrap.resolver.impl.maven.embedded;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.logging.Logger;
+import org.arquillian.spacelift.Spacelift;
+import org.arquillian.spacelift.execution.ExecutionException;
+import org.arquillian.spacelift.task.archive.UntarTool;
+import org.arquillian.spacelift.task.archive.UnzipTool;
+
+import static org.jboss.shrinkwrap.resolver.impl.maven.embedded.DistributionStageImpl.MAVEN_TARGET_DIR;
+
+class FileExtractor {
+
+   private final File fileToExtract;
+   private final File destinationDir;
+   private Logger log = Logger.getLogger(FileExtractor.class.getName());
+
+   FileExtractor(File fileToExtract, String destinationPath) {
+      this.fileToExtract = fileToExtract;
+      this.destinationDir = Paths.get(MAVEN_TARGET_DIR, destinationPath).toFile();
```

Comment:
Please, change the name of the `destinationPath` parameter or move the 
`Paths.get(MAVEN_TARGET_DIR, destinationPath).toFile();` 
to the `DistributionStageImpl.java` class (I would vote for it). Otherwise, it's quite confusing that the parameter name is `destinationPath`, but it is just a subdirectory name